tuning/trainercontroller/callback.py

```diff
@@ -50,11 +51,73 @@
 CONTROLLER_METRICS_KEY = "controller-metrics"
 OPERATIONS_KEY = "operations"
 CONTROLLERS_KEY = "controllers"
+ARGS_KEY = "arguments"
 
 CONTROLLER_NAME_KEY = "name"
-CONTROLLER_TRIGGERS_KEY = "triggers"
 CONTROLLER_RULE_KEY = "rule"
-CONTROLLER_OPERATIONS_KEY = "operations"
+CONTROLLER_TRIGGERS_KEY = "triggers"
+CONTROLLER_OPERATIONS_KEY = OPERATIONS_KEY
+
+CONTROLLER_METRICS_SCHEMA_YAML = f"""
+$schema: "https://json-schema.org/draft/2020-12/schema"
+$id: "https://example.com/product.schema.json"
+title: "Metrics and Operations"
+description: "Controller metrics and operations."
+type: object
+additionalProperties: false
+properties:
+    "{CONTROLLER_METRICS_KEY}":
+        type: array
+        items:
+            type: object
+            required: ["name", "class"]
```

Comment:
I see you are reusing the variables `*_KEY` to infix values in to the schema string. Do you also want to do that for required constraint fields as well?


```suggestion
            required: ["{CONTROLLER_NAME_KEY}", "class"]
```

Similar changes wherever needed.